packages/web-config-server/src/apiV1/dataBuilders/generic/orgUnit/facilityOpeningHours.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { addRow } from './addRow';
+
+const SORTED_WEEKDAYS = [
+  'Monday',
+  'Tuesday',
+  'Wednesday',
+  'Thursday',
+  'Friday',
+  'Saturday',
+  'Sunday',
+];
+
+export const facilityOpeningHours = async ({ dataBuilderConfig, query }, aggregator) => {
+  const { dataElementCodes: rawDataElementCodes, dataServices } = dataBuilderConfig;
+  const dataElementCodes = Object.values(rawDataElementCodes).flat();
+  const { results } = await aggregator.fetchAnalytics(dataElementCodes, { dataServices }, query);
+  const resultsCodeToValue = Object.fromEntries(results.map(row => [[row.dataElement], row.value]));
+
+  const openingHours = {};
+  Object.entries(rawDataElementCodes).forEach(([day, codes]) => {
+    // If it is open this day 'codes[0]', and has opening time 'codes[1]' and closing time 'codes[2]'
+    if (
+      codes[0] &&
+      resultsCodeToValue[codes[0]] === 'Yes' &&
+      codes[1] &&
+      resultsCodeToValue[codes[1]] &&
+      codes[2] &&
+      resultsCodeToValue[codes[2]]
+    ) {
+      openingHours[day] = `${resultsCodeToValue(codes[1])} - ${resultsCodeToValue(codes[2])}`;
+    } else openingHours[day] = 'Closed';
```

Comment:
The logic here doesn't cover the following cases accurately:

**Case A**:
Answers for eg Monday:
* Is Open: (not answered)

It will display `Closed`, will we should display `No data` or similar

**Case B**:
* Is Open: Yes
* Opening time: (not answered)
* Closing time: (not answered)

It will display `Closed`, which contradicts the "Is Open" answer. We should display `Open`, `Open (unknown times)` or similar - please check with Georgia.

Reply:
One way to mitigate the above issues is by making the "Is Open" question mandatory, and the "Opening/Closing" time questions mandatory if "Is Open" is Yes, but this will

1. Require survey updates
2. Again, code best practice would be to not depend on those survey config details

Reply:
Thanks Kostas. That is very conprehensive logic. Georgia mentioned that `opening and closing hours` should come following  the `isOpen`, but I agree it would be better to add some input checks.